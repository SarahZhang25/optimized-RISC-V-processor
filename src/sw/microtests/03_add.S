// This tests that your processor handles ADD (and ADDI) correctly.
.text
start:
    addi x1, x0, 1
    add x2, x1, x0
    add x3, x2, x1
    add x4, x3, x2
    add x5, x4, x3
    add x6, x5, x4
    addi x7, x0, 0x7FF
    add x8, x1, x7
    add x9, x7, x7
    addi x10, x0, -2
    add x11, x1, x10
    unimp
