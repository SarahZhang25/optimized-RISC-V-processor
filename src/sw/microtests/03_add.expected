Dumping the state of the processor
pc = 0x0000002c
x0 = 0x00000000
x1 = 0x00000001
x2 = 0x00000001
x3 = 0x00000002
x4 = 0x00000003
x5 = 0x00000005
x6 = 0x00000008
x7 = 0x000007ff
x8 = 0x00000800
x9 = 0x00000ffe
x10 = 0xfffffffe
x11 = 0xffffffff
x12 = 0x00000000
x13 = 0x00000000
x14 = 0x00000000
x15 = 0x00000000
x16 = 0x00000000
x17 = 0x00000000
x18 = 0x00000000
x19 = 0x00000000
x20 = 0x00000000
x21 = 0x00000000
x22 = 0x00000000
x23 = 0x00000000
x24 = 0x00000000
x25 = 0x00000000
x26 = 0x00000000
x27 = 0x00000000
x28 = 0x00000000
x29 = 0x00000000
x30 = 0x00000000
x31 = 0x00000000
Quitting simulation.
